app/src/androidTest/java/com/github/polypoly/app/network/RemoteDBTest.kt

```diff
@@ -204,4 +209,268 @@ class RemoteDBTest: PolyPolyTest(false, false) {
         val dataFound = remoteDB.getValue<User>(key).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
         assertEquals(data, dataFound)
     }
+
+    // =============================================================== REMOVE VALUE
+    @Test
+    fun unregisteredDataCantBeRemoved() {
+        val key = "some_key"
+        assertFalse(
+            remoteDB.removeValue(key).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
+        )
+    }
+
+    @Test
+    fun registeredDataCanBeRemoved() {
+        val key = "some_key"
+        val data = TEST_USER_1
+        remoteDB.registerValue(key, data).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
+        assertTrue(
+            remoteDB.removeValue(key).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
+        )
+    }
+
+    @Test
+    fun registeredDataIsRemoved() {
+        val key = "some_key"
+        val data = TEST_USER_1
+        remoteDB.registerValue(key, data).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
+        remoteDB.removeValue(key).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
+        assertFalse(
+            remoteDB.keyExists(key).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
+        )
+    }
```

Comment:
Those two tests could be merged into one as they are very close and test small highly related and interdependent behaviors

Reply:
The first checks that `removeValue `returns correctly true, the second one that it is indeed removed, even if that means the same thing, two different things are checked and thus having two tests is a better approach.